x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/index.tsx

```diff
@@ -39,14 +45,42 @@ import {
   policyResponseAppliedRevision,
 } from '../../store/selectors';
 import { EndpointDetails } from './endpoint_details';
+import { EndpointActivityLog } from './endpoint_activity_log';
 import { PolicyResponse } from './policy_response';
+import * as i18 from '../translations';
 import { HostMetadata } from '../../../../../../common/endpoint/types';
+import {
+  EndpointDetailsFlyoutTabs,
+  EndpointDetailsTabsTypes,
+} from './components/endpoint_details_tabs';
+
 import { PreferenceFormattedDateFromPrimitive } from '../../../../../common/components/formatted_date';
 import { EndpointIsolateFlyoutPanel } from './components/endpoint_isolate_flyout_panel';
 import { BackToEndpointDetailsFlyoutSubHeader } from './components/back_to_endpoint_details_flyout_subheader';
 import { FlyoutBodyNoTopPadding } from './components/flyout_body_no_top_padding';
 import { getEndpointListPath } from '../../../../common/routing';
 
+const contentLoadingMarkup = (
```

Comment:
Instead of a singleton, can you either convert this to a function (or Component) or defined inside of `EndpointDetailsFlyout` using `useMemo()`.

Just not sure that we want to reuse the same JSX across multiple instances of `EndpointDetailsFlyout`

Reply:
done e38e202e81f8dd2796bebe9069175d31052c52f2